### PR TITLE
add more tls support

### PR DESCRIPTION
**Pull request by livius-ungureanu**
The current support for op-rabbit allows to enable simple TLS only for dev/test mode(i.e. the underlying trust manager is a TrustEverythingTrustManager).

To overcome this, the op-rabbit's ConnectionParams constructor will gain at the end a new optional SSLContext with default value None. In this way, the backward compatibility is also provided.

By offering the op-rabbit client the possibility to specify  its own `SSLContext`, we decouple the client's specific config(i.e.  the client app might have already setup it's own keystore/truststore manager and also already initialized a SSLContext based on these) from op-rabbit connection config and hence it's more flexible.

The `ConnectionParams` remains backward compatible by adding a 
```
sslContextOpt: Option[SSLContext] = None
```
right at the end and also keeps the constructor coherent:

if ssl is true
- if sslContextOpt is Some(sslContext) then the `factory`  uses the client  `sslContext` 
- otherwise,  `factory` uses the default `SSLcontext` (i.e based on the underlying TrustEverythingTrustManager)

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import java.util.Properties
 
 val json4sVersion = "3.6.6"
-val circeVersion = "0.12.3"
+val circeVersion = "0.12.0-M3"
```
Comment (timcharper) [inline build.sbt]: What's the motivation for this change?

File: build.sbt [hunk 2]
```diff
@@ -16,7 +16,7 @@ val assertNoApplicationConf = taskKey[Unit]("Makes sure application.conf isn't p
 val commonSettings = Seq(
   organization := "com.spingo",
   version := appProperties.getProperty("version"),
-  scalaVersion := "2.12.10",
+  scalaVersion := "2.11.8",
```
Comment (timcharper) [inline build.sbt]: 2.11 ? :)

File: build.sbt (hunk 1 above)
Comment (livius-ungureanu) [reply build.sbt]: Hi Tim
Yes, reverted to the original  `circeVersion`.

File: build.sbt (hunk 2 above)
Comment (livius-ungureanu) [reply build.sbt]: Yes, reverted to the original  `scalaVersion`.

Comment (livius-ungureanu) [reply build.sbt]: Can you check once again please?

**Comment by livius-ungureanu** [comment]
Hi @timcharper 

I have fixed the versions. Can you please take a look?

**Review by timcharper — APPROVED** [review]
